activerecord/lib/active_record/connection_adapters/column.rb

```diff
@@ -17,8 +17,9 @@ class Column
       # +default+ is the type-casted default value, such as +new+ in <tt>sales_stage varchar(20) default 'new'</tt>.
       # +sql_type_metadata+ is various information about the type of the column
       # +null+ determines if this column allows +NULL+ values.
-      def initialize(name, default, sql_type_metadata = nil, null = true, default_function = nil, collation: nil, comment: nil, **)
+      def initialize(name, cast_type, default, sql_type_metadata = nil, null = true, default_function = nil, collation: nil, comment: nil, **)
```

Comment:
I seems like this change breaks oracle enhanced adapter... for whoever is concerned. So far I'm not sure whether it would be a simple matter of updating oracle enhanced adapter to obtain and pass that value or there are deeper logic changes to be addressed.

Reply:
I assume you mean [this gem](https://github.com/rsim/oracle-enhanced).

That's a bug that you should report to the maintainer of that gem; it needs to be updated for the latest ActiveRecord including this refactoring.